packages/rulesets/generated/spectral/az-arm.js

```diff
@@ -2672,7 +2672,7 @@ const ruleset = {
         },
         DeleteResponseCodes: {
             description: "Synchronous DELETE must have 200 & 204 return codes and LRO DELETE must have 202 & 204 return codes.",
-            severity: "error",
+            severity: "off",
```

Comment:
Is severity strict? In other words could set set this to a value like staging? See my comment in your other PR https://github.com/Azure/azure-openapi-validator/pull/531#discussion_r1213756664.

If we cannot do that is there some way to add a tag or keyword of some sort extra metadata in this configuration to identify this as staging?

Reply:
I think we cannot set it to `staging`, per [my comment here](https://github.com/Azure/azure-openapi-validator/pull/531/files/6ab2d5a466c4844216f17e9acdfc3ff2bdd41d01#r1213765233).

I identified it as staging by adding a comment in the source of truth typescript file: [az-arm.ts](https://github.com/Azure/azure-openapi-validator/pull/532/files#diff-4c1382203db84bcd9df61a5bbf90823d0e1f39a833e8eaa1a5be96ca4a4e9b61). The file you are commenting on, `az-arm.js`, is generated from that `.ts` file upon `rush build`.

Reply:
Yes but can that comment be turned into some other tag field we can filter on?

Another possibility is to split out the staging rules into their own rule set list. Instead of being in az-arm.ts they could be az-staging.ts and then we can include or not include that set based on the -is-staging flag at the higher level at, https://github.com/Azure/azure-openapi-validator/blob/main/packages/rulesets/src/index.ts#L10. That way the promotion procedure is to move from az-staging.js to az-staging.ts.

Reply:
> Yes but can that comment be turned into some other tag field we can filter on?

I am unsure about that, because from what I see there is a type conversion going on, from the ruleset yaml into a type provided by the spectral library, [here](https://github.com/Azure/azure-openapi-validator/blob/6ab2d5a466c4844216f17e9acdfc3ff2bdd41d01/packages/azure-openapi-validator/autorest/src/spectral-plugin-func.ts#L205):
`return new Ruleset(ruleset, { severity: "recommended" })`

> Another possibility is to split out the staging rules into their own rule set list. 

I see some complications here. For example, we would need separate staging rules for arm and data plane, and then promoting the rules from staging would mean moving it them over to a different file, resulting in large diff.

How about we stick to the comment for now and iteratively improve the process later on? Note we have a draft PR where we are working on coming up with the rule promotion strategy, here:
- https://github.com/Azure/azure-openapi-validator/pull/518

Reply:
There has to be a more structured approach than relying on a comment and then force turning on all rules as errors when we want to run staging. I'm sure we can figure out something that will allow us to programmatically determine the set of rules that are in staging.

Reply:
Changes applied. Ready for review. Details in:
- #531